src/Datadog.Trace.ClrProfiler.Managed.Core/InsertFirstInterceptMethodAttribute.cs

```diff
@@ -0,0 +1,26 @@
+// <copyright file="InsertFirstInterceptMethodAttribute.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// Attribute that indicates that the decorated method is meant to intercept calls
+    /// to another method. Used to generate the integration definitions file.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
+    public class InsertFirstInterceptMethodAttribute : InterceptMethodAttribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InsertFirstInterceptMethodAttribute"/> class.
+        /// </summary>
+        public InsertFirstInterceptMethodAttribute()
+        {
+            MethodReplacementAction = MethodReplacementActionType.InsertFirst;
```

Comment:
I know we have a unit test for it, but should we make the `MethodReplacementAction` setter `private`/`protected`, so you can't use it directly anyway? (Same goes for `Kind` in the other attribute)

Reply:
Unfortunately, we can't change `DuckAttribute.Kind` because it exists in `Datadog.Trace.dll` which could be referenced by an older `Datadog.Trace.ClrProfiler.Managed.dll` if the NuGet package is used, and several integrations set the `Kind` named argument 😞

I can and will change `MethodReplacementAction` since that type is defined in `Datadog.Trace.ClrProfiler.Managed.Core` and will always be in sync with the profiler because it's provided by our automatic instrumentation packages.

Great suggestion @andrewlock  👍🏼